### ISSUE TITLE
Matlab  syntax highlighting in Spyder editor

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1157]
_From [adeela...@gmail.com](https://code.google.com/u/114902358077470304967/) on 2012-09-29T23:47:25Z_

Spyder Version:  2.1.11
Python Version:  2.7.1 
Qt Version:      4.7.2, PyQt4 (API v1) 4.8.3 on Linux

Hi all,

I am using spyder for more than one year and it (along with numpy,pylab) has enabled me to most of my tasks without using Matlab. (Thanks Spyder Dev. Team)

Recently i was trying to convert some of my old Matlab code into Python code. For this i have to open Matlab-files in Matlab's editor because currently Spyder donot support m-files syntax highlighting.

I think it will be a good feature to add Matlab-syntax highlighting support in Spyder-Editor.

Thanks,

  Adeel

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1157_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-09-30T01:57:24Z_

This issue was closed by revision b161f6909278 .

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [adeela...@gmail.com](https://code.google.com/u/114902358077470304967/) on 2012-09-30T10:20:37Z_

Thanks Pierre for adding this feature so quickly. 

I installed the Spyder using latest revision. Matlab files now opens and comments and strings are ok but some default editor features are somehow not working properly. e.g.

1: In Python/C++ files 'numbers' are always shown in red color (see attached Number_python image) but in *.m file same numbers are black (see attached Number_Matlab)

2: Editor is recognizing Matlab built-in commands e.g. ones, length, reshape etc and these built-in commands are appeared bold (darker see attached Matlab_commands.png) 
 but in Python/C++ files they appears in "Pink" color (see Python_commands.png)


Thanks,

 Adeel

**Attachment:** [Number_Python.png Number_Matlab.png Matlab_commands.png Python_commands.png](http://code.google.com/p/spyderlib/issues/detail?id=1157#hc2)

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-09-30T13:10:04Z_

This issue was updated by revision 8f09a74bf656 .


The MATLAB syntax highlighter is based on a generic syntax highlighter powered
by the Python library 'Pygments', whereas the Python syntax highlighter was
written specifically for Spyder (it's 2 to 3 times faster than would be a
Pygments based Python syntax highlighter). That's why there are differences in
terms of how things are colored. I fixed the builtins color. However, the fact
that the numbers aren't colored is apparently a Pygments bug (Pygments does not
find 'Number' tokens in a string containing numbers, and it's apparently
specific to the MATLAB lexer: Pygments Python lexer is detecting 'Number' tokens
just fine) that I should report to the Pygments dev team.
